### PR TITLE
Deploy example site to GitHub pages

### PR DESCRIPTION
**Pull request by dholbach**
Fixes: #10

### CODE REVIEW

**Comment by dholbach** [comment]
We would need to enable GitHub pages for this. To try this out, I published to my own GH here: https://dholbach.github.io/spectral/ (https://github.com/dholbach/spectral/runs/4121856524?check_suite_focus=true)

File: layouts/partials/stylesheets.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {{ $css := resources.Get "css/main.css" | minify | fingerprint "sha512" }}
-<link rel="stylesheet" href="{{ $css.RelPermalink }}" integrity="{{ $css.Data.Integrity }}">
+<link rel="stylesheet" href="{{ $css.Permalink }}" integrity="{{ $css.Data.Integrity }}">
```
Comment (dholbach) [inline layouts/partials/stylesheets.html:2]: Unfortunately this was the only way to make this work reliably with the `baseUrl` being `a.b.c/d`

Comment (sbruder) [reply layouts/partials/stylesheets.html:2]: When I tested it with a baseURL ending in `/spectral/`, hugo automatically made the RelPermalinks begin with `/spectral/`. Is there a case where that is not sufficient?

**Comment by sbruder** [comment]
I created a new `gh-pages` branch and GitHub Pages seems to be enabled, so the GitHub Action should work after merging.

File: layouts/partials/stylesheets.html (hunk 1 above)
Comment (dholbach) [reply layouts/partials/stylesheets.html]: When I try it locally it works too - for some reason it didn't, when I used the GH Workflow. Maybe let's just drop the change for now.

**Comment by dholbach** [comment]
Let's land this now and fix issues if they come up later on.

**Comment by dholbach** [comment]
→ https://sbruder.github.io/spectral/

**Pull request merged**